erpnext/selling/report/territory_wise_sales/territory_wise_sales.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from erpnext import get_default_currency
+from frappe import _
+
+def execute(filters=None):
+	filters = frappe._dict(filters)
+	columns = get_columns()
+	data = get_data(filters)
+	return columns, data
+
+
+def get_columns():
+	currency = get_default_currency()
+	return [
+		{
+			"label": _("Territory"),
+			"fieldname": "territory",
+			"fieldtype": "Link",
+			"options": "Territory"
+		},
+		{
+			"label": _("Opportunity Amount"),
+			"fieldname": "opportunity_amount",
+			"fieldtype": "Currency",
+			"options": currency
+		},
+		{
+			"label": _("Quotation Amount"),
+			"fieldname": "quotation_amount",
+			"fieldtype": "Currency",
+			"options": currency
+		},
+		{
+			"label": _("Order Amount"),
+			"fieldname": "order_amount",
+			"fieldtype": "Currency",
+			"options": currency
+		},
+		{
+			"label": _("Billing Amount"),
+			"fieldname": "billing_amount",
+			"fieldtype": "Currency",
+			"options": currency
+		}
+	]
+
+def get_data(filters=None):
+	data = []
+
+	opportunities = get_opportunities(filters)
+	quotations = get_quotations(opportunities)
+	sales_orders = get_sales_orders(quotations)
+	sales_invoices = get_sales_invoice(sales_orders)
+
+	for territory in frappe.get_all("Territory"):
+		territory_opportunities = list(filter(lambda x: x.territory == territory.name, opportunities)) if opportunities and opportunities else []
```

Comment:
territory_opportunities = []
if opportunities:
         territory_opportunities = list(filter(lambda x: x.territory == territory.name, opportunities))
         t_opportunity_names = [t.name for t in territory_opportunities] if territory_opportunities else []